packages/compiler-cli/linker/test/ast/babel/babel_ast_host_spec.ts

```diff
@@ -0,0 +1,282 @@
+          .toThrowError('Unsupported syntax, expected element in array not to be empty.');
+    });
+
+    it('should error if there is a spread element', () => {
+      expect(() => host.parseArrayLiteral(expr(template.ast`[...[0,1]]`)))
+          .toThrowError('Unsupported syntax, expected element in array not to use spread syntax.');
+    });
+  });
+
+  describe('isObjectLiteral()', () => {
+    it('should return true if the expression is an object literal', () => {
+      expect(host.isObjectLiteral(rhs(template.ast`x = {}`))).toBe(true);
+      expect(host.isObjectLiteral(rhs(template.ast`x = { foo: 'bar' }`))).toBe(true);
+    });
+
+    it('should return false if the expression is not an object literal', () => {
+      expect(host.isObjectLiteral(rhs(template.ast`x = "moo"`))).toBe(false);
+      expect(host.isObjectLiteral(rhs(template.ast`x = 'moo'`))).toBe(false);
+      expect(host.isObjectLiteral(rhs(template.ast`x = someIdentifier`))).toBe(false);
+      expect(host.isObjectLiteral(rhs(template.ast`x = 42`))).toBe(false);
+      expect(host.isObjectLiteral(rhs(template.ast`x = []`))).toBe(false);
+      expect(host.isObjectLiteral(rhs(template.ast`x = null`))).toBe(false);
+      expect(host.isObjectLiteral(rhs(template.ast`x = 'a' + 'b'`))).toBe(false);
+      expect(host.isObjectLiteral(rhs(template.ast`x = \`moo\``))).toBe(false);
+    });
+  });
+
+  describe('parseObjectLiteral()', () => {
+    it('should extract the properties from the object', () => {
+      const moo = expr(template.ast`'moo'`);
+      expect(host.parseObjectLiteral(rhs(template.ast`x = {}`))).toEqual(new Map());
+      expect(host.parseObjectLiteral(rhs(template.ast`x = {a: 'moo'}`)))
+          .toEqual(new Map([['a', moo]]));
+    });
+
+    it('should error if there is a method', () => {
+      expect(() => host.parseObjectLiteral(rhs(template.ast`x = { foo() {} }`)))
+          .toThrowError('Unsupported syntax, expected a property assignment.');
+    });
+
+    it('should error if there is a spread element', () => {
+      expect(() => host.parseObjectLiteral(rhs(template.ast`x = {...{a:'moo'}}`)))
+          .toThrowError('Unsupported syntax, expected a property assignment.');
+    });
+  });
+
+  describe('isFunctionExpression()', () => {
+    it('should return true if the expression is a function', () => {
+      expect(host.isFunctionExpression(rhs(template.ast`x = function() {}`))).toBe(true);
+      expect(host.isFunctionExpression(rhs(template.ast`x = function foo() {}`))).toBe(true);
+      expect(host.isFunctionExpression(rhs(template.ast`x = () => {}`))).toBe(true);
+      expect(host.isFunctionExpression(rhs(template.ast`x = () => true`))).toBe(true);
+    });
+
+    it('should return false if the expression is a function declaration', () => {
+      expect(host.isFunctionExpression(expr(template.ast`function foo() {}`))).toBe(false);
+    });
+
+    it('should return false if the expression is not an array literal', () => {
```

Comment:
I see, you copy-pasted all of it! ✂️
```suggestion
    it('should return false if the expression is not a function expression', () => {
```

Reply:
(maybe...)